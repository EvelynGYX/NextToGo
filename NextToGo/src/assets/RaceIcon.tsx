import React from 'react';
import {StyleSheet} from 'react-native';
import FontAwesome5Icon from 'react-native-vector-icons/FontAwesome5';
import Category from '../constants/Category';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

interface RaceItemIconProps {
  category: Category;
}

const RaceIcon = (props: RaceItemIconProps) => {
  return (
    <>
      {props.category === Category.Horse ? (
        <FontAwesome5Icon name="horse-head" size={20} style={styles.icon} />
      ) : props.category === Category.Greyhound ? (
        <FontAwesome5Icon style={styles.icon} name="dog" size={20} />
      ) : (
        <MaterialCommunityIcons
          style={styles.icon}
          name="horse-human"
          size={20}
        />
      )}
    </>
  );
};

const styles = StyleSheet.create({
  icon: {
    paddingRight: 24,
  },
});
export default RaceIcon;
