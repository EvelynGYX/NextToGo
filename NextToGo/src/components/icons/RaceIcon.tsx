import React from 'react';
import {StyleProp, ViewStyle} from 'react-native';
import FontAwesome5Icon from 'react-native-vector-icons/FontAwesome5';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import Category from '../../constants/Category';

export interface RaceItemIconProps {
  category: Category;
  style?: StyleProp<ViewStyle>;
}

const RaceIcon = (props: RaceItemIconProps) => {
  return (
    <>
      {props.category === Category.Horse ? (
        <FontAwesome5Icon style={props.style} name="horse-head" size={20} />
      ) : props.category === Category.Greyhound ? (
        <FontAwesome5Icon style={props.style} name="dog" size={20} />
      ) : (
        <MaterialCommunityIcons
          style={props.style}
          name="horse-human"
          size={20}
        />
      )}
    </>
  );
};

export default RaceIcon;
