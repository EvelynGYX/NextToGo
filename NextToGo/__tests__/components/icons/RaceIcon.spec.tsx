import React from 'react';
import {StyleProp, ViewStyle} from 'react-native';
import renderer from 'react-test-renderer';
import RaceIcon from '../../../src/components/icons/RaceIcon';
import Category from '../../../src/constants/Category';

describe('RaceIcon', () => {
  const mockStyle: StyleProp<ViewStyle> = {height: 100};
  const mockPropsArray: [string, StyleProp<ViewStyle>, Category][] = [
    ['Horse', mockStyle, Category.Horse],
    ['Greyhound', mockStyle, Category.Greyhound],
    ['Harness', mockStyle, Category.Harness],
  ];

  it.each(mockPropsArray)(
    'renders correctly with style props if category is %s ',
    (_, styles, category) => {
      const tree = renderer
        .create(<RaceIcon style={styles} category={category} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    },
  );
});
