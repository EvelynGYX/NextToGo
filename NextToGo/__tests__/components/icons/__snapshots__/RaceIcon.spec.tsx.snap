// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaceIcon renders correctly with style props if category is Greyhound  1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 20,
      },
      Object {
        "height": 100,
      },
      Object {
        "fontFamily": "FontAwesome5Free-Solid",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {
        "fontWeight": "900",
      },
    ]
  }
>
  
</Text>
`;

exports[`RaceIcon renders correctly with style props if category is Harness  1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 20,
      },
      Object {
        "height": 100,
      },
      Object {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  󱗀
</Text>
`;

exports[`RaceIcon renders correctly with style props if category is Horse  1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 20,
      },
      Object {
        "height": 100,
      },
      Object {
        "fontFamily": "FontAwesome5Free-Solid",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {
        "fontWeight": "900",
      },
    ]
  }
>
  
</Text>
`;
